### ISSUE TITLE
Sheriff reorders the fields alphabetically

### ISSUE THREAD

**Issue by skhattane** [liip/sheriff#49]
The usage of `sheriff.Marshal(options, data)` reorganizes the fields alphabetically instead of keeping the original order. (v2.0.0-beta.1)

Expected
```
{
    "id": "a5b002bb-b63d-4192-b665-ec4dadcb036b",
    "name": "test",
    "createdAt": "2024-03-17T06:56:04Z",
    "updatedAt": "2024-03-17T06:56:04Z"
}
```

Actual behavior
```
{
    "createdAt": "2024-03-17T06:56:04Z",
    "id": "a5b002bb-b63d-4192-b665-ec4dadcb036b",
    "name": "test",
    "updatedAt": "2024-03-17T06:56:04Z"
}
```

**Comment by Chalwe19** [comment]
+1

**Comment by mweibel** [comment]
`sheriff.Marshal()` uses as a target a `map[string]interface{}` to achieve the flexibility it needs to do it's job. `json.Marshal` sorts maps. I'm not aware of a way to keep the original order. 

If you know a way, please propose a solution and we can talk about a PR. Or submit a PR directly but depending on the complexity I may decline to merge it. 

Reasoning behind is that while it may be slightly annoying, JSON is a format mostly read by machines and not by humans and for machines order does not matter.

**Comment by mweibel** [comment]
FYI this is fixed with release v2.0.0-beta.2. See README on instructions for adding ordered map output.
